src/libstore/http-binary-cache-store.cc

```diff
@@ -14,6 +14,9 @@ struct HttpBinaryCacheStoreConfig : virtual BinaryCacheStoreConfig
 
     const std::string name() override { return "HTTP Binary Cache Store"; }
 
+    const Setting<bool> negotiate{this, false, "negotiate",
+        "Whether to do kerberos negotiate when talking to the http binary cache."};
```

Comment:
Does this need to be a configuration item, or could it be autodetected?

I think `negotiate` is a bit vague. Would `kerberos` be a better name?

Reply:
We could add `CURLAUTH_ANY` instead of `CURLAUTH_NEGOITATE`, which would do the detection for you. We could add that flag at all times, and libcurl will use the best available authentication it finds. This is what git does, and that works pretty well. The downside is that if the HTTP server will accept negotiation OR another protocol, libcurl will always choose negotiation, which has a very slim chance of breaking use cases. This may be a very acceptable risk as git has been doing this for a decade already.

`negotiate` is the best name here, it matches the curl command line argument `--negotiate`, and is widely known as that.

Reply:
I'm not too familiar with that, but it's generally good to mirror existing interfaces.
What about `CURLAUTH_ANYSAFE `?
Maybe we should similarly expose a single `auth` setting, instead of specific ones for each possible solution.

Reply:
A decent idea, I've created #10584 as an alternate proposal to this one.